File: algorithmProblemSolving/programmers/level2/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

bool solution(vector<string> phone_book) {
    if(phone_book.size()==1){
        return true;
    }//벡터에 전화번호가 하나만 있는 경우 배제
    else{
        sort(phone_book.begin(), phone_book.end());
        for(int i=0;i<phone_book.size()-1;i++){
            if(phone_book[i]==phone_book[i+1].substr(0,phone_book[i].size())) return false;
        }
    }
    return true;
}

File: algorithmProblemSolving/programmers/level2/전화번호 목록.py
def solution(phone_book):
    answer = True
    phone_book.sort()
    print(phone_book)
    if len(phone_book) == 1 :
        return answer
    else:
        for i in range(len(phone_book)-1):
            if phone_book[i] == phone_book[i+1][0:len(phone_book[i])]:
                answer = False
                return answer
    return answer